### PR TITLE
[feat|sde-backend] : Policy-hub Proxy controller and Policy pojo added

### PR DESCRIPTION
**Pull request by ChetanT-System**
### Added
- Added Policy-hub Proxy controller.
- Added New Policy Entities and pojo classes.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/********************************************************************************
+ * Copyright (c) 2022, 2023 T-Systems International GmbH
+ * Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation
```
Comment (sachinargade123) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java:3]: 2024 Copy right missing

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/SubmodelJsonReq.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/********************************************************************************
+ * Copyright (c) 2022, 2023 T-Systems International GmbH
+ * Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation
```
Comment (sachinargade123) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/SubmodelJsonReq.java:3]: please check all files for 2024 copyright

**Review by sachinargade123 — CHANGES_REQUESTED** [review]
Please verify all files for 2024 copyright

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/SubmodelJsonReq.java (hunk 2 above)
Comment (ChetanT-System) [reply modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/SubmodelJsonReq.java]: Updated

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java (hunk 1 above)
Comment (ChetanT-System) [reply modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java]: Updated

**Review by sachinargade123 — APPROVED** [review]
LTGM

**Review by adityagajbhiye9 — APPROVED** [review]
LGTM

**Comment by adityagajbhiye9** [comment]
@almadigabor  Please review and merge.

**Review by sachinargade123 — CHANGES_REQUESTED** [review]
Don't put old year copy right files for new Files like 2022, 2023

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/********************************************************************************
+ * Copyright (c) 2022, 2023, 2024 T-Systems International GmbH
```
Comment (sachinargade123) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java:2]: This file is new, it contains old copy right year

Comment (ChetanT-System) [reply modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/PolicyModel.java]: Updated all copyrights headers. Please review again

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/validators/ValidationService.java [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /********************************************************************************
- * Copyright (c) 2022, 2023 T-Systems International GmbH
- * Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation
+ * Copyright (c) 2022, 2023, 2024 T-Systems International GmbH
+ * Copyright (c) 2022, 2023, 2024 Contributors to the Eclipse Foundation
```
Comment (dvasunin) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/validators/ValidationService.java:3]: Look at [Copyright Headers](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) of the Handbook. There is no need to include all years, you may include only the year in which the content was created. Alternatively, you can keep year when a file was updated. Intermediate years are not needed

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/CommonPropEntity.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+	@Column(name = "deleted")
+	@JsonProperty(value = "deleted")
+	private String deleted;
+
+	@Column(name = "updated")
+	@JsonProperty(value = "updated")
+	private String updated;
+
+}
```
Comment (dvasunin) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/entities/CommonPropEntity.java:76]: I am doubting about having JSON and persistence-related annotations been used in a single source file

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/mapper/AspectResponseFactory.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import lombok.SneakyThrows;
+
+@Component
+public class AspectResponseFactory {
+
+	ObjectMapper mapper = new ObjectMapper();
+
```
Comment (dvasunin) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/mapper/AspectResponseFactory.java:36]: where is the mapper used? Usually, it's not a good idea to mix fasterxml and Gson libraries altogether

**Review by dvasunin — CHANGES_REQUESTED** [review]

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/mapper/AspectResponseFactory.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+package org.eclipse.tractusx.sde.common.mapper;
+
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
```
Comment (dvasunin) [inline modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/mapper/AspectResponseFactory.java:29]: Usually it is not a good idea to mix Gson and fasterxml

**Review by dvasunin — APPROVED** [review]
The proposed changes can be considered later on refactoring. For now LGTM

**Comment by adityagajbhiye9** [comment]
merge before #87

**Comment by ChetanT-System** [comment]
@almadigabor  We have updated copyright header with current year whichever class has affected/ modified.

File: modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/validators/ValidationService.java (hunk 4 above)
Comment (almadigabor) [reply modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/validators/ValidationService.java:3]: Good point @dvasunin! Although I have to emphasise the you have to use either one year which is when the file was created (this is the one you mentioned) or 2 years where the first is the creation year and second is the last modification year.

Comment (ChetanT-System) [reply modules/sde-common/src/main/java/org/eclipse/tractusx/sde/common/validators/ValidationService.java]: Updated as per Handbook

**Comment by ChetanT-System** [comment]
@almadigabor Updated copyright headers as per handbook

**Review by almadigabor — APPROVED** [review]
LGTM

**Pull request merged**